[BE-3192]: Added test report comp doc, improved ios versioning and ad… (#451)
appcircleio/appcircle-docusaurus: docs/workflows/ios-specific-workflow-steps/test-reports-for-ios.md

---
title: Test Reports for iOS
description: Test Report step displays your test results and code coverage in an aesthetically pleasing user interface.
tags: [test, test report, test result, automation]
sidebar_position: 18
---

import Screenshot from '@site/src/components/Screenshot';

# Test Reports for iOS

The Appcircle **Test Report** step displays your test results and code coverage in an aesthetically pleasing user interface.

This component supports the following test and coverage formats:

> - [**Xcode 13+ XCTest**](https://developer.apple.com/documentation/xctest)
> - [**JUnit**](https://junit.org)
> - [**Cobertura**](https://cobertura.github.io/cobertura)
> - [**lcov.info**](https://lcov-viewer.netlify.app)

Reviewer: ```suggestion
- [Xcode 13+ XCTest](https://developer.apple.com/documentation/xctest) - For Apple's native test framework.
- [JUnit](https://junit.org) - For Java-based test reporting.
- [Cobertura](https://cobertura.github.io/cobertura) - For coverage reporting.
- [lcov.info](https://lcov-viewer.netlify.app) - For GCC coverage data.
```

For additional details, please refer to the document: [**Generating Test Report**](/continuous-testing/ios-testing/running-ios-unit-and-ui-tests#generating-test-report)

### Prerequisites

Before executing the **Test Report** workflow step, certain prerequisite workflow steps must be completed:

| Prerequisite Workflow Step                                   | Description                                                                   |
| ------------------------------------------------------------ | ----------------------------------------------------------------------------- |

> | [**Xcodebuild for Unit and UI Test**](/workflows/ios-specific-workflow-steps/xcodebuild-for-unit-and-ui-test) | This step allows you to run unit and UI tests in your project. When this step runs, all your tests are run, and an .xcresult file is created as a result. |

Reviewer: ```suggestion
| [**Xcodebuild for Unit and UI Test**](/workflows/ios-specific-workflow-steps/xcodebuild-for-unit-and-ui-test) | Run unit and UI tests in your project to generate an `.xcresult` file containing the test outcomes. |
```

### Input Variables

For each component, specific input variables are required for its operation on your system. 

| Variable Name            | Description                                                      | Status    |
| ------------------------ | ---------------------------------------------------------------- | --------- |

> | `AC_TEST_RESULT_PATH`    | Specifies the directory and its subdirectories where compatible test files will be searched. | Required  |

Reviewer: ```suggestion
| `AC_TEST_RESULT_PATH`    | Define the directory and its subdirectories for searching compatible test files. | Required  |
```

> | `AC_COVERAGE_RESULT_PATH`| This environment variable is automatically set for native iOS projects when you run tests. For other projects, you need to set the coverage path. | Optional  |

Reviewer: ```suggestion
| `AC_COVERAGE_RESULT_PATH`| For native iOS projects, tests automatically set this variable. For other projects, you must specify the coverage path manually. | Optional  |
```
